File: 2020/02/part2.c
#include <stdio.h>
#include <string.h>

int main() {
	FILE *fp = fopen("data.txt", "r");
	int min, max;
	char c;
	char password[100];
	int count = 0;
	while(fscanf(fp, "%d-%d %c: %s\n", &min, &max, &c, password) == 4) {
		int match = 0;
		if(password[min - 1] == c) {
			match++;
		}
		if(password[max - 1] == c) {
			match++;
		}
		if(match == 1) {
			count++;
		}
	}
	printf("%d\n", count);
	return 0;
}

File: Y2023/D02/part2.js
const fs = require('fs')

const input = fs.readFileSync(`${__dirname}/data.txt`, 'utf8').split('\n')

let total = 0;
for(const line of input) {
    let [_, _game, a] = line.match(/Game (\d+): (.+)/);
    a = a.split('; ').map(t => t.split(',').map(s => s.trim()).map(s => s.split(' ')).reduce((acc, s) => ({ ...acc, [s[1]]: +s[0] }), {}));
    const max = {
        red: 1,
        green: 1,
        blue: 1
    }
    for(const set of a) {
        for(const color of Object.keys(max)) {
            if(set[color] > max[color]) {
                max[color] = set[color];
            }
        }
    }
    total += max['red'] * max['green'] * max['blue'];
}

console.log(total);